src/plugins/telemetryTable/TelemetryTable.js

```diff
@@ -206,17 +208,62 @@ define([
 
         addColumnsForObject(telemetryObject) {
             let metadataValues = this.openmct.telemetry.getMetadata(telemetryObject).values();
-
             metadataValues.forEach(metadatum => {
                 let column = this.createColumn(metadatum);
                 this.configuration.addSingleColumnForObject(telemetryObject, column);
+                // add units column if available
+                if(metadatum.unit !== undefined) {
+                    let unitColumn = this.createUnitColumn(metadatum);
+                    this.configuration.addSingleColumnForObject(telemetryObject, unitColumn);
+                }
             });
         }
 
         createColumn(metadatum) {
             return new TelemetryTableColumn(this.openmct, metadatum);
         }
 
+        createUnitColumn(metadatum) {
+            return new TelemetryTableUnitColumn(this.openmct, metadatum);
+            // let unitColumn = {
+            //     isUnit: true,
+            //     metadatum,
+            //     titleValue: metadatum.name + ' Unit',
+            //     selectable: false,
+            //     formatter: {
+            //         format(telemetryDatum) {
+            //             return metadatum.unit;
+            //         },
+            //         parse(telemetryDatum) {
+            //             return metadatum.unit;
+            //         }
+            //     },
+            //     getKey() {
+            //         return this.metadatum.key + '-unit';
+            //     },
+            //     getTitle() {
+            //         return this.titleValue;
+            //     },
+            //     getMetadatum() {
+            //         return this.metadatum;
+            //     },
+            //     hasValueForDatum(telemetryDatum) {
+            //         return telemetryDatum.hasOwnProperty(this.metadatum.source);
+            //     },
+            //     getRawValue(telemetryDatum) {
+            //         return this.metadatum.unit;
+            //     },
+            //     getFormattedValue(telemetryDatum) {
+            //         return this.formatter.format(telemetryDatum);
+            //     },
+            //     getParsedValue(telemetryDatum) {
+            //         return this.formatter.parse(telemetryDatum);
+            //     }
+            // };
+            // return unitColumn;
```

Comment:
If this isn't instructional we should remove this

Reply:
Yeah that was before I created that new unit column class!